src/contexts/AuthContext.jsx

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useContext, createContext } from "react";
+import { useNavigate } from "react-router-dom";
+
+const AuthContext = createContext(null);
+
+export const useAuth = () => useContext(AuthContext);
+
+export const AuthProvider = ({ children }) => {
+  const navigate = useNavigate();
+
+  const [auth, setAuth] = useState({
+    userId: localStorage.getItem("userId"),
+    isLoggedIn: JSON.parse(localStorage.getItem("isLoggedIn")),
+    encodedToken: localStorage.getItem("encodedToken"),
+  });
+
+  useEffect(() => {
+    localStorage.setItem("userId", auth.userId);
+    localStorage.setItem("isLoggedIn", auth.isLoggedIn);

```

Comment:
Store only one json stringified object in localstorage as user keep isLoggedIn in reducer state just a suggestion

Reply:
Did to this way for easy access and to keep each item independent of each other